lib/augmentConfig.js

```diff
@@ -402,10 +402,9 @@ function addProcessorFunctions(config /*: stylelint$config*/) /*: stylelint$conf
 			if (processorCache.has(processorKey)) {
 				initializedProcessor = processorCache.get(processorKey);
 			} else {
-				/** @type {[string, Object]} */
 				processorConfig = /** @type {Array<string | Object>} */ ([]).concat(processorConfig);
-				const processorLookup = processorConfig[0];
-				const processorOptions = processorConfig[1];
+				const processorLookup = /** @type {[string, Object]} */ (processorConfig)[0];
```

Comment:
I noticed that you add parenthesis, where they not needed and no one adds them there, from my experience:

```diff
- processorConfig[0]
+ (processorConfig)[0]
```

```diff
- [].concat()
+ ([]).concat()
```

Is there a reason for this? It makes code a little bit harder to understand, because it's unusual.

Reply:
`/** @type {any} */` applies to first statement in brackets. It is sort of casting types in typescript in jsdoc, see https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#casts